pkg/tinydtls/contrib/sock_dtls.c

```diff
@@ -461,4 +550,65 @@ static inline uint32_t _update_timeout(uint32_t start, uint32_t timeout)
     return (diff > timeout) ? 0: timeout - diff;
 }
 
+#ifdef SOCK_HAS_ASYNC
+void _udp_cb(sock_udp_t *udp_sock, sock_async_flags_t flags, void *ctx)
+{
+    sock_dtls_t *sock = ctx;
+
+    if (flags & SOCK_ASYNC_MSG_RECV) {
+        sock_dtls_session_t remote;
+        void *data = NULL;
+        void *data_ctx = NULL;
+
+        ssize_t res = sock_udp_recv_buf(udp_sock, &data, &data_ctx, 0,
+                                        &remote.ep);
+        if (res <= 0) {
+            DEBUG("sock_dtls: error receiving UDP packet: %d\n", (int)res);
+            return;
+        }
+
+        if (sock->buf_ctx != NULL) {
```

Comment:
Why the check for `sock->buf_ctx`?

Reply:
But a value is assigned to it. If it is already set, we can't store something else in it. The `DEBUG()` below basically says as much.